CRM/Civisocial/Backend/OAuthProvider/Facebook.php

```diff
@@ -0,0 +1,243 @@
+<?php
+require_once 'CRM/Civisocial/Backend/OAuthProvider.php';
+
+class CRM_Civisocial_Backend_OAuthProvider_Facebook extends CRM_Civisocial_Backend_OAuthProvider {
+
+	/**
+	 * Short name (alias) for OAuth provider
+	 *
+	 * @var string
+	 */
+	private $alias = 'facebook';
+
+	/**
+	 * Construct Facebook OAuth object
+	 *
+	 * @param string $accessToken
+	 *		Preobtained access token. Makes the OAuth Provider ready
+	 *		to make requests.
+	 */
+	public function __construct($accessToken = NULL) {
+		$this->apiUri = 'https://graph.facebook.com/v2.6';
+		$this->getApiCredentials($this->alias);
+		$this->token = $accessToken;
+	}
+
+	/**
+	 * Authorization URI that user will be redirected to for login
+	 *
+	 * @param array $permissions
+	 *		Permissions to be requested
+	 *
+	 * @return string | bool
+	 * @todo Check if requests have been reviewed by Facebook
+	 */
+	public function getLoginUri($permissions = array(), $reRequest = FALSE) {
+		$uri = 'https://www.facebook.com/dialog/oauth';
+		$params = array(
+			'client_id' => $this->apiKey,
+			'redirect_uri' => $this->getCallbackUri($this->alias),
+		);
+		if (empty($permissions)) {
+			$params['scope'] = implode(',', array_merge($this->getBasicPermissions(), $this->getExtraPermissions()));
+		} else {
+			$params['scope'] = implode(',', $permissions);
```

Comment:
I wrote that thinking a scenario like this. Civisocial needs to post on the user's timeline but it doesn't have `publish_action`  permission. So, asking only that permission as additional made more sense. Because at this particular instance we only need publish permission. Basic permissions are already assumed to have been granted. See this line. https://github.com/rajbdilip/org.civicrm.civisocial/blob/22ce540d31cd1fc111e689868d08316b6d9ccee7/CRM/Civisocial/Backend/OAuthProvider/Facebook.php#L111

But yeah. Appending basic permissions will be better in case if they have revoked any of the basic permissions. :+1: :)